packages/babel-plugin/src/utils/__tests__/css-builders.test.ts

```diff
@@ -0,0 +1,122 @@
+import generate from '@babel/generator';
+import { parse } from '@babel/parser';
+import type { NodePath } from '@babel/traverse';
+import traverse from '@babel/traverse';
+import type { Identifier, MemberExpression } from '@babel/types';
+
+import type { Metadata } from '../../types';
+import { buildCss } from '../css-builders';
+
+describe('buildCss', () => {
+  it('returns a css string and variables array for an identifier node', () => {
+    const file = parse(`
+      const color = { background: 'red' };
+      const styles = color;
+
+      run(styles);
+    `);
+
+    let path: NodePath<Identifier> | null = null;
+    traverse(file, {
+      CallExpression(nodePath) {
+        nodePath.traverse({
+          Identifier(innerPath) {
+            if (innerPath.node.name === 'styles') {
+              path = innerPath;
+            }
+          },
+        });
+      },
+    });
+
+    expect(path).not.toEqual(null);
+    const meta: Metadata = {
+      parentPath: path!.parentPath,
+      state: {
+        filename: '',
+      },
+    } as any;
+
+    const { css, variables } = buildCss(path!.node, meta);
+    expect(css).toEqual([{ css: 'background: red;', type: 'unconditional' }]);
+    expect(variables).toEqual([]);
+  });
+
+  it('returns a css string and variables array for a member expression node', () => {
+    const file = parse(`
+      const styles = { option1: { background: 'red' } };
+
+      run(styles.option1);
+    `);
+
+    let path: NodePath<MemberExpression> | null = null;
+    traverse(file, {
+      CallExpression(nodePath) {
+        nodePath.traverse({
+          MemberExpression(innerPath) {
+            path = innerPath;
+          },
+        });
+      },
+    });
+
+    expect(path).not.toEqual(null);
+
+    const meta: Metadata = {
+      parentPath: path!.parentPath,
+      state: {
+        cssMap: {},
+        filename: '',
+      },
+    } as any;
+
+    const { css, variables } = buildCss(path!.node, meta);
+    expect(css).toEqual([{ css: 'background: red;', type: 'unconditional' }]);
+    expect(variables).toEqual([]);
+  });
+
+  it('returns a css string for a variable member expression', () => {
+    const file = parse(
+      `
+      import { css } from '@compiled/react';
+
+      const styles = { option1: css({ background: 'red' }) };
+
+      run(styles[key]);
+    `,
+      { sourceType: 'module' }
+    );
+
+    let path: NodePath<MemberExpression> | null = null;
+    traverse(file, {
+      CallExpression(nodePath) {
+        nodePath.traverse({
+          MemberExpression(innerPath) {
+            path = innerPath;
+          },
+        });
+      },
+    });
+
+    expect(path).not.toEqual(null);
+
+    const meta: Metadata = {
+      parentPath: path!.parentPath,
+      state: {
+        cssMap: {},
+        filename: '',
+      },
+    } as any;
+
+    const { css, variables } = buildCss(path!.node, meta);
+    // TODO: This should not happen
```

Comment:
isn't this a bug? i'm not sure why we're adding a test case to enforce buggy behaviour here - perhaps creating a github issue would be better than having this test case here, as the former is more easily discoverable (we don't really check back on test cases unless they get broken)

the other test cases in this file look great, thank you for adding them 🙌

Reply:
Sure, let me remove some of these; generally I'm only trying to reproduce bugs with these.

It's a bit of a structural issue regarding what `css` + `cssMap` APIs do.